src/Gui/Placement.cpp

```diff
@@ -559,6 +572,83 @@ void Placement::showDefaultButtons(bool ok)
     }
 }
 
+bool Placement::eventFilter(QObject*, QEvent* ev) {
+    //handle spacenav daemon motion events in placement window
+    if (//event->type() == Spaceball::ButtonEvent::ButtonEventType ||
+        ev->type() == Spaceball::MotionEvent::MotionEventType){
+                Spaceball::MotionEvent *motionEvent = dynamic_cast<Spaceball::MotionEvent*>(ev);
+                if (!motionEvent) {
+                    Base::Console().Log("invalid spaceball motion event in bool Placement::eventFilter(QObject*, QEvent* ev)\n");
+                    return false;
+                    }
+                motionEvent->setHandled(true);
+
+                static double translationConstant(.001f);
+                double xTrans = static_cast<double>(motionEvent->translationX())*translationConstant;
+                double yTrans = static_cast<double>(motionEvent->translationY())*translationConstant;
+                double zTrans = static_cast<double>(motionEvent->translationZ())*translationConstant;
+
+                static double rotationConstant(.001f);
+                double dY=static_cast<double>(motionEvent->rotationZ()) * rotationConstant;
+                double dP=static_cast<double>(motionEvent->rotationY()) * rotationConstant;
+                double dR=static_cast<double>(motionEvent->rotationX()) * rotationConstant;
+
+                if(xTrans == 0 && yTrans == 0 && zTrans == 0 && dY ==0 && dP ==0 && dR ==0 ) return false;
+                signalMapper->blockSignals(true);
+//                printf("\nSpaceball Placement Event\t(x,y,z)=(%f,%f,%f)\t(Rx,Ry,Rz)=(%f,%f,%f);\n",xTrans,yTrans,zTrans,dR,dP,dY  );
+
+                Base::Vector3d dTrans(xTrans,yTrans,zTrans);
+                Base::Vector3d rotationCenter=this->getCenterData();
+                Base::Placement old_placement(getPositionData(),getRotationData());
+                // Code below allows for seamles rotation around axes without blocking on roll, pitch, yaw range ends.
+                Base::Matrix4D new_placement_mat;
+                new_placement_mat = old_placement.toMatrix();
+                new_placement_mat.move(-rotationCenter);
+                if(dY!=0) new_placement_mat.rotZ(Base::toRadians(dY));
+                if(dP!=0) new_placement_mat.rotY(Base::toRadians(dP));
+                if(dR!=0) new_placement_mat.rotX(Base::toRadians(dR));
+                new_placement_mat.move(rotationCenter);
+                new_placement_mat.move(dTrans);
+                Base::Placement new_placement(new_placement_mat);
+                new_placement.setPosition(old_placement.getPosition()+dTrans);
+                //setPlacementData(new_placement);//Does not work properly because of problems with euler angles. Angle values need to be adjusted below to fit into proper range.

```

Comment:
What to do with this?

Reply:
At first I was trying to minimize count of UI updates but the first approach did not work properly. 
It was supposed to be a note explaining why the rest of the algorithm was needed.
Can be removed or maybe rewritten in hopefully more clear way.
Which way would be preferred?